packages/discord.js/src/structures/User.js

```diff
@@ -188,12 +204,32 @@ class User extends Base {
   }
 
   /**
-   * The Discord "tag" (e.g. `hydrabolt#0001`) for this user
+   * The tag of this user
+   * <info>This user's username, or `hydrabolt#0001` if they're using the legacy username system</info>
    * @type {?string}
    * @readonly
+   * @deprecated Use {@link User#username} instead.
    */
   get tag() {
-    return typeof this.username === 'string' ? `${this.username}#${this.discriminator}` : null;
+    if (!tagDeprecationEmitted) {
+      process.emitWarning('User#tag is deprecated. Use User#username instead.', 'DeprecationWarning');
+      tagDeprecationEmitted = true;
+    }
+
+    return typeof this.username === 'string'
+      ? this.discriminator === '0'
+        ? this.username
+        : `${this.username}#${this.discriminator}`
+      : null;
+  }
+
+  /**
+   * The global name of this user, or their username if they don't have one
+   * @type {?string}
```

Comment:
Can this actually be nullable?

Reply:
Yup, in fact your `global_name` is `null`:
![image](https://github.com/discordjs/discord.js/assets/24852502/e4510da6-7cb8-4435-8a6f-c176c3b5d510)

Reply:
I know global_name can be null, I'm talking about displayName

Reply:
Right, I didn't see the lines below sorry.

Seeing it falls back to `username` which has never been `null`, this getter can't be nullable indeed.
```suggestion
   * @type {string}
```

Reply:
Yes, if the user is partal